Automattic/wp-calypso
# Plans 2023: Show 2yr cost savings as a price discount

Related to #74261 
Based off https://github.com/Automattic/wp-calypso/pull/74206

## Proposed Changes

Closes #74261

Shows the cost savings between 2-year plans and their 1-year variant in a discounted fashion. When visiting `/plan/2yearly/[SITE]` the (maybe discounted) price for the yearly plan variant is shown crossed-out, with the biennial plan's (maybe discounted) price shown below e.g. with a site on a free plan:

<img width="700" alt="Screenshot 2023-03-17 at 3 54 21 PM" src="https://user-images.githubusercontent.com/1705499/225924887-398ce263-e40f-43bd-a0a0-6628335b4a9f.png">

### Notes/Concerns

Some edge cases or thoughts/concerns that may need addressing before moving on with this:

1. We seem to have already different "forms" of discount displays, and this adds another one to the mix. This may introduce ambiguity for anyone exposed to more than one form. To my knowledge, there are currently (in `trunk`) currency-based discounts and pro-rated credits discounts that are displayed similarly. This PR introduces another form, "cost-savings" (from 1yr to 2yr plans). It might be the case we need to communicate a little better what is being discounted here (maybe a subscript or "information" bubble next to the price to clarify - more below).
2. The output when visiting `/plans/2yearly/[SITE]` with a site on a paid plan and pro-rated credits available (the Commerce plan's to be specific):

<img width="700" alt="Screenshot 2023-03-20 at 12 49 06 PM" src="https://user-images.githubusercontent.com/1705499/226318265-20231e70-120c-493a-933b-c2c4edfa0a0f.png">

^ AFIAK these are the correct prices for a site on the Business plan. The `20` vs `23.50` discrepancy for the Commerce plan is not a calculation error (same prices when viewing `/plans/1yearly` or `/plans/2yearly` in `trunk`). It will probably be confusing to someone viewing this grid without additional context ("why do we pay more for the biennial version"?). The answer is they are not - it's just different periods being compared. The price for the yearly plan is `(540 - 300 credits) / 12 months = 20` and for biennial `(864 - 300 credits) / 24 months = 23.50`. It's almost like comparing apples to oranges here. We can probably handle this a little better, some thoughts: 

- add an informational bubble or subscript clarifying, something like _"(for the _first_ year. with pro-rated credits applied)"_ - or even be a little more "geeky" and show the actual formula used. 
- maybe project the 1-year plan prices over a 2-year period, so we are comparing same periods? e.g. the yearly plan price would become `( 1080 - 300 credits ) / 24 months = 32.50`

^ In the same image above, the price for the Business plan (current plan) is shown as `25 vs 7.50`, which might be a little confusing / need additional clarification. 

## Testing Instructions



* Go to `/plans/[SITE]` with the feature flag enabled `?flags=plans/biannual-toggle`
* Switch to the 2-yearly grid
* Confirm that the prices and sums are correct and shown in a discounted fashion as above.
* Confirm with both a free plan and a paid plan that gives our "pro-rated" credits used towards an upgrade (so the grid forces discounted instead of the raw prices to show) e.g. a site on Business plan
* Confirm with different currencies e.g. IRN to ensure those discounts are also visible in the prices

## Pre-merge Checklist
